// node-csfd-api/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.csfd = exports.Csfd = void 0;
const creator_service_1 = require("./services/creator.service");
const movie_service_1 = require("./services/movie.service");
const search_service_1 = require("./services/search.service");
const user_ratings_service_1 = require("./services/user-ratings.service");
class Csfd {
    constructor(userRatingsService, movieService, creatorService, searchService) {
        this.userRatingsService = userRatingsService;
        this.movieService = movieService;
        this.creatorService = creatorService;
        this.searchService = searchService;
    }
    async userRatings(user, config) {
        return this.userRatingsService.userRatings(user, config);
    }
    async movie(movie) {
        return this.movieService.movie(+movie);
    }
    async creator(creator) {
        return this.creatorService.creator(+creator);
    }
    async search(text) {
        return this.searchService.search(text);
    }
}
exports.Csfd = Csfd;
const movieScraper = new movie_service_1.MovieScraper();
const userRatingsScraper = new user_ratings_service_1.UserRatingsScraper();
const creatorScraper = new creator_service_1.CreatorScraper();
const searchScraper = new search_service_1.SearchScraper();
exports.csfd = new Csfd(userRatingsScraper, movieScraper, creatorScraper, searchScraper);
